File: go_practice/sentinel.go
package main

import "fmt"

func main() {
	arr := []int{1, 2, 3, 4}
	var num int
	i := 0
	fmt.Scan(&num)
	arr = append(arr, num)
	for num != arr[i] {
		i += 1
	}
	if i == (len(arr) - 1) {
		fmt.Println("見つかりませんでした")
	} else {
		fmt.Printf("%dに見つかりました", i)
	}
}

File: c_practice/sentinel.c
#include <stdio.h>

int main(void)
{
  int arr[5] = {1, 3, 6, 8};
  int num, i = 0;
  scanf("%d", &num);
  arr[((sizeof(arr) / sizeof(arr[0])) - 1)] = num;
  while (num != arr[i])
  {
    i += 1;
  }
  if (i < 5 - 1)
    printf("%d番目にある\n", i);
  else
    printf("見つかりませんでした\n");
  return 0;
}
